src/raw/owned_bson.rs

```diff
@@ -0,0 +1,504 @@
+use std::convert::{TryFrom, TryInto};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    oid::{self, ObjectId},
+    raw::RAW_BSON_NEWTYPE,
+    spec::ElementType,
+    Binary,
+    Bson,
+    DbPointer,
+    Decimal128,
+    RawArray,
+    RawArrayBuf,
+    RawBinary,
+    RawBson,
+    RawDbPointer,
+    RawDocument,
+    RawDocumentBuf,
+    RawJavaScriptCodeWithScope,
+    RawRegex,
+    Regex,
+    Timestamp,
+};
+
+use super::{
+    serde::{OwnedOrBorrowedRawBson, OwnedOrBorrowedRawBsonVisitor},
+    Error,
+    Result,
+};
+
+/// A BSON value backed by owned raw BSON bytes.
+#[derive(Debug, Clone, PartialEq)]
+pub enum OwnedRawBson {
```

Comment:
This type was needed to have something to append to documents. I originally tried using a `RawBson<'a>` for this, but it was clunky / awkward, and it would potentially make the `rawdoc!` macro impossible.

Regarding the name, I originally opted for `RawBson` being the owned one and `RawBsonRef` being the borrowed one, but I ended up going with `OwnedRawBson` and `RawBson` to match how we added the suffix to the owned variants of `RawDocument` and `RawArray`. I'm still not certain this is the correct naming scheme for `RawBson` though, would love to hear your all's thoughts on this.

Reply:
I think I agree. I originally thought it would be confusing to have `RawDocument` typically be borrowed and `RawBson` be owned, but `OwnedRawBson` does seem odd enough to warrant the inconsistency. Also, `RawDocument` _technically_ isn't a borrowed type anyways, though it's always accessed as a wide-pointer (`&RawDocument`).

Reply:
Updated to be `RawBsonRef` and `RawBson`. I also updated `RawBinary<'a>` et al to be `RawBinaryRef<'a>` and what not. `OwnedRawJavaScriptCodeWithScope` dropped the `Owned` prefix. I also renamed `RawBsonRef::to_owned_raw_bson` to just `RawBsonRef::to_raw_bson`.